### DISCUSSION TITLE
Getting "Uncaught Promise {<pending>}" console errors but everything works...

### DISCUSSION THREAD

**Discussion by vivianeasley** [pmndrs/valtio#131] [Q&A]
First off, if you are a maintainer of this library, it's amazing! Completely what I've been looking for, for a long time now. 

I've started to replace a similar proxy based state management library in my React project with this one. In my app I'm making a request using Axios and returning a promise and setting it on state using Valatio. I'm using React suspense and everything works great but I keep getting console errors that say `Uncaught Promise {<pending>}`. I can see where this is getting emitted in Valatio (lines 335-340) but I'm a bit unsure what I'm doing wrong. I don't necessarily need a fix as much as just maybe an explanation to why the error is getting thrown if someone knows. Thank you in advance!

**Comment by dai-shi** [comment]
Great to hear that!

Can you create a minimal reproduction of the console error with codesandbox?
I think I never saw such errors while playing with examples.

**Reply by vivianeasley** [reply]
So I spent some time today trying to get the error in codesandbox and couldn't but I was using roughly the same code. In the end I moved the request into the component and the error when away. I could clearly see that I had been trying to set a pending promise on state using Valatio and that was causing the error. When I made sure the request had resolved before setting it on state everything worked perfectly. I think I just didn't 100% understand how promises should interact with the library given what I read on the github page documentation. Honestly, just seeing [this](https://github.com/pmndrs/valtio/discussions/91) kind of helped me understand its usage a little better. Anyways, thanks again for such a great state management library! I'm really enjoying it!

**Reply by dai-shi** [reply]
I'd admit the current readme doesn't cover well about async usage.
The truth is it's because we don't know best practice yet.
Would you share the codesandbox even if it doesn't repro the error?
We can learn people's expectations and then possibly improve the readme.
